CONTRIBUTING.md

```diff
@@ -25,6 +25,13 @@ Avoid exception handling. Because Hyrise is not a product, we do not have to rec
 - Use forward declarations whenever possible to reduce compile time
 - We mostly use structs for PODS (plain old data structures). If it has methods, chances are that it is a class.
 
+## Using smart pointers and references
+- If an object is stored in a smart pointer, pass that pointer around as a const reference, e.g., `const std::shared_ptr<Table> &`
```

Comment:
I believe the most important exception is missing:
> The exception to this rule is when the called function does something that removes that "Instance up in the stack"

In a simple call, this is safe, because the stack above us will ensure that the shared_ptr remains valid. As soon as we store this smart pointer somewhere, we have to store an actual smart pointer, not a reference to the object or a reference to the smart pointer.

Maybe add another rule:

> - Neither smart pointers nor the objects they point to may be stored as a reference. As an example: In a method `MyClass::set_table(const std::shared_ptr<Table>& table)`, you can freely use the reference, because the caller will take care of the shared pointer. Once you store `table` in `MyClass`, you must store it as an `std::shared_ptr<Table>`. This is so that the pointer remains valid even if `set_table` and the methods above it are no longer on the stack.

Reply:
updated. is that better?